spec/features/invoices/show_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+describe 'invoices show page', js: true do
+  include MoneyHelper
+
+  let(:user) { create :user }
+  let(:org)  { create :organization, with_user: user }
+
+  before do
+    sign_in user
+  end
+
+  subject{ page }
+
+  context 'Download as PDF' do
+    let!(:invoice) { create :invoice, organization: org }
+
+    context 'Download' do
+      before do
+        visit invoice_path(invoice)
+        click_on 'Download as PDF'
+      end
+
+      it { expect(page.response_headers['Content-Type']).to eq 'application/pdf' }
+      it { expect(page.response_headers['Content-Disposition']).to \
+        include("#{invoice.customer.to_s}_#{invoice.ends_at.month}_#{invoice.ends_at.year}.pdf") }
```

Comment:
@yunixon how about to move this logic of generating PDF file name to model? Also please fix it in invoices_controller.